internal/tools/aws/database.go

```diff
@@ -224,3 +248,41 @@ func (d *RDSDatabase) rdsDatabaseProvision(installationID string, logger log.Fie
 
 	return nil
 }
+
+func (d *RDSDatabase) getKMSResourceNames(awsID string) ([]*string, error) {
+	kmsResources, err := d.client.resourceTaggingGetAllResources(resourcegroupstaggingapi.GetResourcesInput{
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(DefaultRDSEncryptionTagKey),
+				Values: []*string{aws.String(awsID)},
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get KMS resources with tag %s:%s", DefaultRDSEncryptionTagKey, awsID)
+	}
+
+	resourceNameList := make([]*string, len(kmsResources))
+	for i, resource := range kmsResources {
+		resourceNameList[i] = resource.ResourceARN
+	}
+
+	return resourceNameList, nil
+}
+
+func (d *RDSDatabase) getEnabledEncryptionKeys(resourceNameList []*string, logger log.FieldLogger) []*kms.KeyMetadata {
+	var keys []*kms.KeyMetadata
+
+	for _, name := range resourceNameList {
+		keyMetadata, err := d.client.kmsGetSymmetricKey(*name)
+		if err != nil {
+			logger.WithError(err).Error("Failed get some of the encryption key")
+			continue
+		}
```

Comment:
Only logging an error here and not returning seems dangerous. What if there are 1 or more keys that would have been returned that would have altered the rest of the flow?

Reply:
That's a good point. I will change it. Thanks!